### ISSUE TITLE
Incorrect esc_html usage within add_note method in class-edd-payment

### ISSUE THREAD

**Issue by davidsherlock** [awesomemotive/easy-digital-downloads#6967]
### Expected behavior
When notes are automatically recorded, such as by Commissions on refunded payments, the html/links are broken due to the incorrect usage of `esc_html` within the class-edd-payment class. This should be `wp_kses` instead, as per the `edd_ajax_store_payment_note` function (https://github.com/easydigitaldownloads/easy-digital-downloads/blob/3d5c8e9269d10b7e972c94d5acf7c5ea838019d1/includes/admin/payments/actions.php#L331)

### Actual behavior

Note outputs like this:

```
admin – October 2, 2018, 8:03 pm
Commission revoked for admin due to refunded payment – <a href="https://themedd.local/wp-admin/edit.php?post_type=download&page=edd-commissions&payment=1478">View</a&gt; – Delete
```

### Steps to reproduce the behavior

1) Enable commissions and the "Revoke on refund" option
2) Refund a payment and check the nots
3) The a href/URL is outputted directly within the note, instead of being a clickable.

### Information (if a specific version is affected):
PHP Version: 7.2

EDD Version (or branch): 2.9.8

WordPress Version: Version 4.9.8

**Comment by davidsherlock** [comment]
For reference, customer ticket: https://github.com/easydigitaldownloads/easy-digital-downloads/issues/6967

**Comment by Spreeuw** [comment]
note that using `wp_kses` in the same way as the frontend AJAX handler would strip the links altogether (with the second argument being an empty array in this particular case). It appears that some of the functions here are used backwards.

Codex [suggests](https://codex.wordpress.org/Data_Validation#HTML.2FXML_Fragments) kses for **input** before saving to the DB:
> Note that the kses system can be resource-intensive, and should therefore not be run as an output sanitization filter directly, but as a filter to data after it has been input and processed, before it is saved in the database. WordPress runs kses on the pre_comment_content filter, for example, to filter the HTML before saving the comment.

and esc_* for output ([codex](https://codex.wordpress.org/Validating_Sanitizing_and_Escaping_User_Data#Escaping:_Securing_Output)):
> For security on the other end of the spectrum, we have escaping. To escape is to take the data you may already have and help secure it prior to rendering it for the end user. WordPress thankfully has a few helper functions we can use for most of what we'll commonly need to do:
> 
> esc_html() we should use anytime our HTML element encloses a section of data we're outputting.

That first link to the kses bit is quite confusing though because it lists kses under the output header and then explains that it is best used for input sanitization. I can see how that ended up being the root cause of this issue (seeing the lenghty discussions in #3926.
Currently (I believe as a result of that pull request), the note is `esc_html`'d before input and `kses`'ed before output rather than vice versa.

**Comment by Spreeuw** [comment]
confirmed `release/3.0` is affected as well

**Comment by cklosowski** [comment]
Basically we'll need to rethink how we handle all these calls since they all end up piping to a single place...

I don't think this is a single fix but a more standardization of the process of adding notes.

**Comment by JJJ** [comment]
> confirmed `release/3.0` is affected as well

3.0 is only affected on the way in, where all HTML is completely stripped from notes. 3.0 does not use `esc_html()` on the way out, but rather calls `make_clickable()` to ensure that links are formatted in a desirable way, inside the new `edd_admin_get_note_html()` function. (We could add more auto-formatting here, but I don't think it's worth it quite yet.)

Will push a few improvements to `release/3.0` shortly to help clarify some of this in the code.

**Comment by Spreeuw** [comment]
3.0 does use `esc_html()` on the way in: 
https://github.com/easydigitaldownloads/easy-digital-downloads/blob/release/3.0/includes/payments/class-edd-payment.php#L1740

I wrote this little test to quickly get different notes in a payment:
``` php
$payment_id = 1;
$note = '<a href="https://google.com">Google</a>';
// this is a plain direct insertion, results in proper HTML on the frontend
edd_insert_payment_note( $payment_id, "edd_insert_payment_note(): ".$note );
// this is an insertion done via the EDD_Payment object, as is done by EDDC
$payment = new EDD_Payment( $payment_id );
$payment->add_note( "EDD_Payment: ".$note );
// this is an insertion done like the ajax insertion via the backend
edd_insert_payment_note( $payment_id, "Frontend AJAX: ".wp_kses( $note, array() ) );
```
Not sure if 3.0 ajax still uses wp_kses, function appears to have moved.
And the results in 3.0 were the same as in 2.9 (the screenshot doesn't show it but the first link works):
![edd-3 0-note-test](https://user-images.githubusercontent.com/3217773/46502690-0ea97100-c829-11e8-8e1d-9130cf51a3dc.png)

**Comment by JJJ** [comment]
> 3.0 does use esc_html() on the way in:

Only in that 1 place (that I couldn't find, so thanks!) which is definitely not intended, and is a bug/holdover from EDD2.9. I'll fix that now in `release/3.0`.

**Comment by cklosowski** [comment]
This appears resolved in 3.0 as things like affiliate referrals are linked properly instead of the markup being output.
